ReSharper.FSharp/src/FSharp.Psi.Features/src/Parsing/FSharpImplTreeBuilder.fs

```diff
@@ -1036,11 +1036,14 @@ type FSharpExpressionTreeBuilder(lexer, document, lifetime, path, projectedOffse
         | SynExpr.YieldOrReturnFrom(_, expr, _) ->
             x.PushRangeAndProcessExpression(expr, range, ElementType.YIELD_OR_RETURN_EXPR)
 
-        | SynExpr.LetOrUseBang(_, _, _, pat, expr, ands, inExpr, range, _) ->
+        | SynExpr.LetOrUseBang(_, isUse, _, pat, expr, ands, inExpr, range, _) ->
             x.PushRange(range, ElementType.LET_OR_USE_EXPR)
             x.PushExpression(inExpr)
             x.PushStepList(ands, andLocalBindingListProcessor)
-            x.PushRangeForMark(expr.Range, x.Mark(pat.Range), ElementType.LOCAL_BINDING)
+            let letOrUseBangToken = if isUse then FSharpTokenType.USE_BANG else FSharpTokenType.LET_BANG
+            x.AdvanceToTokenOrRangeStart(letOrUseBangToken, range)
+            let exprWithPatRange = Range.mkRange expr.Range.FileName pat.Range.Start expr.Range.End
```

Comment:
`Range.unionRanges` is a simpler way to do it.

Reply:
Fixed. Also, removed unnecessary AdvanceToTokenOrRangeStart because it seems like it is needed only in AND/AND_BANG cases.